ksql-parser/src/test/java/io/confluent/ksql/schema/ksql/TypeContextUtilTest.java

```diff
@@ -0,0 +1,110 @@
+package io.confluent.ksql.schema.ksql;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+import io.confluent.ksql.parser.tree.Array;
+import io.confluent.ksql.parser.tree.Map;
+import io.confluent.ksql.parser.tree.PrimitiveType;
+import io.confluent.ksql.parser.tree.Struct;
+import io.confluent.ksql.parser.tree.Type;
+import io.confluent.ksql.parser.tree.Type.SqlType;
+import io.confluent.ksql.util.KsqlException;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class TypeContextUtilTest {
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  @Test
+  public void shouldGetTypeFromVarchar() {
+    // Given:
+    final String schemaString = "VARCHAR";
+
+    // When:
+    final Type type = TypeContextUtil.getType(schemaString);
+
+    // Then:
+    assertThat(type, is(PrimitiveType.of(SqlType.STRING)));
+  }
+
+  @Test
+  public void shouldGetTypeFromStringArray() {
+    // Given:
+    final String schemaString = "ARRAY<VARCHAR>";
+
+    // When:
+    final Type type = TypeContextUtil.getType(schemaString);
+
+    // Then:
+    assertThat(type, is(Array.of(PrimitiveType.of(SqlType.STRING))));
+  }
+
+  @Test
+  public void shouldGetTypeFromIntArray() {
+    // Given:
+    final String schemaString = "ARRAY<INT>";
+
+    // When:
+    final Type type = TypeContextUtil.getType(schemaString);
+
+    // Then:
+    assertThat(type, is(Array.of(PrimitiveType.of(SqlType.INTEGER))));
+  }
+
+  @Test
+  public void shouldGetTypeFromMap() {
+    // Given:
+    final String schemaString = "MAP<VARCHAR, INT>";
+
+    // When:
+    final Type type = TypeContextUtil.getType(schemaString);
+
+    // Then:
+    assertThat(type, is(Map.of(PrimitiveType.of(SqlType.INTEGER))));
```

Comment:
Unrelated to this PR, but I've never noticed this before: does KSQL only support maps with string keys? The parser's `Map` type only stores a `valueType` and no key type.

Reply:
I believe so - otherwise it is modeled as an array with entries. There's some magic that allows you to do that but I'm not so familiar with it (note that avro does not support non-string keys for maps: https://avro.apache.org/docs/1.8.1/spec.html#Maps)

Reply:
I can provide some context here. The array with entries representation is a part of the avro schema written by the connect converters (not the KSQL schema) to get around the limitations of avro which you've pointed out. We don't support such maps because of a historical limit that's still around from our original avro serdes which didn't use the converters. We can clean up this limitation now.